packages/gazelle_cli/lib/commands/delete/delete_project.dart

```diff
@@ -0,0 +1,41 @@
+import 'dart:io';
+
+/// Uninstalls a Gazelle project.
+class DeletingProjectError {
+  /// The error message.
+  final String message;
+
+  /// Creates a [DeletingProjectError].
+  const DeletingProjectError() : message = "Failed to delete project!";
+}
+
+/// Asks for confirmation before deleting a project.
+///
+/// Returns `true` if user confirms, `false` otherwise.
+Future<bool> getConfirmation(String message) async {
+  print('$message (y/N)');
+  String? answer = stdin.readLineSync();
+  if (answer?.toLowerCase() == 'y' || answer?.toLowerCase() == 'yes') {
+    return true;
+  }
+  return false;
+}
+
+/// Delete a Gazelle project.
+///
+/// Throws a [DeletingProjectError] if project does not exist.
+
+Future<String> deleteProject(
+  String projectName, {
+  String? path,
+}) async {
+  final nameOption = projectName;
+  final pathOption = path;
+  final completePath =
+      pathOption != null ? "$pathOption/$nameOption" : nameOption;
+  if (!await Directory(completePath).exists()) {
+    throw DeletingProjectError();
+  }
+  await Directory(completePath).delete(recursive: true);
+  return completePath;
+}
```

Comment:
Removing the `projectName` parameter should make things easier and simpler.
The function signature would be `Future<String> deleteProject({String? path})`.

The body would also be simpler in my opinion, as you can simplify the first lines with:
```dart
final path = path ?? Directory.current.path;
// ...Continues with deletion  code
```
Does this make sense to you? Let me know what you think!

Reply:
Those changes are done now we don't need nameOption anymore